capellambse/model/__init__.py

```diff
@@ -364,6 +364,44 @@ def by_uuid(self, uuid: str) -> common.GenericElement:
         """Search the entire model for an element with the given UUID."""
         return common.GenericElement.from_model(self, self._loader[uuid])
 
+    def find_references(
+        self, target: ModelObject, /
+    ) -> cabc.Iterator[tuple[ModelObject, str, int | None]]:
+        """Search the model for references to the given object.
+
+        Parameters
+        ----------
+        target
+            The target object to search for.
+
+        Yields
+        ------
+        tuple[ModelObject, str, int | None]
+            A 3-tuple containing the referencing model object, the
+            attribute on that object, and an optional index. If the
+            attribute contains a list of objects, the index shows the
+            index into the list that was found. Otherwise the index is
+            None.
+        """
+        for obj in self.search():
+            for attr in _reference_attributes(type(obj)):
+                if attr.startswith("_"):
+                    continue
+
+                try:
+                    value = getattr(obj, attr)
+                except Exception:
+                    continue
+
+                if isinstance(value, GenericElement) and value == target:
+                    yield (obj, attr, None)
+                elif isinstance(value, ElementList):
+                    try:
+                        idx = value.index(target)
+                    except ValueError:
+                        continue
+                    yield (obj, attr, idx)
```

Comment:
For explanation:
The xpath `f"//*[@*='#{target.uuid}']"` finds all elements which have an attribute named arbitrarily (`*`) and match the UUID of the target in the value. This isn't enough somehow. By using the `LinkAccessor` (which should be named `LinkProxyAccessor` to be precise) an attribute is defined which tracks references on child elements. That is the sole reason we do another for loop `for obj in (nobj, nobj.parent):` which doubles the execution time but is insignificant following the [master-theorem](https://en.wikipedia.org/wiki/Master_theorem_(analysis_of_algorithms)). Here `a=2`.

Now this version executes in less than 30 secs on big projects... and it could be even faster when using just the `find` or `iterfind` of `lxml`.([source: Yellow note](https://lxml.de/xpathxslt.html#xpath))